cmake/modules/root.cmake

```diff
@@ -0,0 +1,49 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2021, Nordic Semiconductor ASA
+
+# Configure default Zephyr roots.
+#
+# This CMake module will set ZEPHYR_BASE as a default root for all supported roots.
+#
+# It will also convert all relative paths in existing ROOT lists to absolute
+# path relative from APPLICATION_SOURCE_DIR.
+#
+# Outcome:
+# The following variables will be defined when this CMake module completes:
+#
+# - DEFAULT_ARCH_ROOT
+# - DEFAULT_BOARD_ROOT
+# - DEFAULT_SOC_ROOT
+# - DEFAULT_MODULE_EXT_ROOT
+# where value is set to ZEPHYR_BASE
```

Comment:
> I don't see the need for these variables, unless the intention is for them to be overridable with a user-supplied cache variable?

I might have been thinking a bit to far ahead in this case.
I agree, let's keep it simple for now, and re-introduce later in case we need this fine grained control.

Reply:
Fixed, and description updated accordingly.